velox/functions/prestosql/StringFunctions.h

```diff
@@ -47,6 +47,55 @@ struct CodePointFunction {
   }
 };
 
+/// trail(string, N) -> varchar
+///
+///     Returns the last N characters of the input string.
+template <typename T>
+struct TrailFunction {
+  VELOX_DEFINE_FUNCTION_TYPES(T);
+
+  // Results refer to strings in the first argument.
+  static constexpr int32_t reuse_strings_from_arg = 0;
+
+  template <typename I>
+  FOLLY_ALWAYS_INLINE void
+  call(out_type<Varchar>& result, const arg_type<Varchar>& input, I N) {
```

Comment:
Can we use `callNullFree` to get default NULL free fast path
https://facebookincubator.github.io/velox/develop/scalar-functions.html#null-free-fast-path